packages/interactivity-router/src/index.ts

```diff
@@ -363,6 +358,62 @@ export const { state, actions } = store( 'core/router', {
 	},
 } );
 
+/**
+ * Announces a message to screen readers.
+ *
+ * This is a wrapper around the `@wordpress/a11y` package's `speak` function. It handles importing
+ * the package on demand and should be used instead of calling `ally.speak` direacly.
+ *
+ * @param messageKey The message to be announced by assistive technologies.
+ * @param ariaLive   The politeness level for aria-live; default: 'polite'.
+ */
+function a11yAnnounce(
+	messageKey: 'loading' | 'loaded',
+	ariaLive?: 'polite' | 'assertive'
+) {
+	if ( ! navigationTexts.loadedFromServer ) {
+		navigationTexts.loadedFromServer = true;
+		const content = document.getElementById(
+			'wp-script-module-data-@wordpress/interactivity-router'
+		)?.textContent;
+		if ( content ) {
+			try {
+				const parsed = JSON.parse( content );
+				if ( typeof parsed?.i18n?.loading === 'string' ) {
+					navigationTexts.loading = parsed.i18n.loading;
+				}
+				if ( typeof parsed?.i18n?.loaded === 'string' ) {
+					navigationTexts.loaded = parsed.i18n.loaded;
+				}
+			} catch {}
+		}
+	}
+
+	const message = navigationTexts[ messageKey ];
+
+	if ( globalThis.IS_GUTENBERG_PLUGIN ) {
+		import( '@wordpress/a11y' ).then(
+			( { speak } ) => speak( message, ariaLive ),
+			// Use the fallback Interactivity API implementation if the a11y
+			// Script Module cannot be loaded.
+			() => {
+				state.navigation.message =
```

Comment:
Here, we would have to keep the corresponding HTML snippet that renders the live aria region.

Reply:
[That's in WordPress Core right now.
](https://github.com/WordPress/wordpress-develop/blob/47122105f20624e5a6b8ed151d732157f2f2605c/src/wp-includes/interactivity-api/class-wp-interactivity-api.php#L988-L1013)

Even if the HTML rendering is removed, continuing to update the state shouldn't be harmful.

Reply:
I'm not convinced in the case when every byte matters 😅 

Before this change:

<img width="653" alt="Screenshot 2024-09-12 at 13 35 25" src="https://github.com/user-attachments/assets/47fa20e3-ca0a-4a38-b3d1-ac465dfd6e6d">

Reply:
Ok, removed.